partition/androidboot.go

```diff
@@ -75,3 +76,13 @@ func (a *androidboot) SetBootVars(values map[string]string) error {
 	}
 	return env.Save()
 }
+
+func (a *androidboot) Reboot() error {
+	// Write argument so we reboot to recovery partition
+	param := []byte("recovery\n")
+	if err := ioutil.WriteFile("/run/systemd/reboot-param", param, 0644); err != nil {
```

Comment:
I prefer to use the same file because this way system-shutdown works properly not only for snapd, but also for the "reboot" command (there are those two ways to generate the file, so better to use the same interface, otherwise we would need to support two different ways of receiving the information in system-shutdown). Note also that we would not need the file if "shutdown +x<time> -r" worked the same way as "reboot", which is a systemd inconsistency.